src/ucp/rndv/rndv_rtr.c

```diff
@@ -133,8 +136,6 @@ ucp_proto_rndv_rtr_data_received_zcopy(ucp_request_t *req, int in_buffer)
 {
     static const unsigned dt_mask = UCS_BIT(UCP_DATATYPE_CONTIG);
 
-    ucp_datatype_iter_mem_dereg(req->send.ep->worker->context,
```

Comment:
i think we should remove ucp_proto_rndv_rtr_data_received_bcopy/ucp_proto_rndv_rtr_data_received_zcopy now, and use ucp_proto_rndv_rtr_common_complete directly, with UCP_DT_MASK_ALL

Reply:
I'd leave it, this is extra protection that recv_zcopy is only relevant for contig dts

Reply:
why is it important to protect this? it's not like there would be a leak or some other error.. also we plan to extend zcopy to iov dt as well

Reply:
to have an assert on an invalid flow.
ok, will unite